pkg/git/git_test.go

```diff
@@ -420,8 +439,17 @@ func (u *UtilsGitMock) plainClone(path string, isBare bool, o *git.CloneOptions)
 	return nil, nil
 }
 
+func (u *UtilsGitMock) plainOpen(path string) (*git.Repository, error) {
+	u.path = path
+	return nil, nil
+}
+
 type UtilsGitMockError struct{}
 
 func (UtilsGitMockError) plainClone(string, bool, *git.CloneOptions) (*git.Repository, error) {
 	return nil, errors.New("error during clone")
 }
+
+func (UtilsGitMockError) plainOpen(path string) (*git.Repository, error) {
```

Comment:
[basically unrelated] Maybe another approach would be to have only one GitMock with a property `thrown` or `thrownOn<functionName>` or something like that. Then we can set `thrown` when instanciating. This gets returned when the method is called. But I see we have the dual mock approach already in the sources. So that basically unrelated and should - if at all - changed maybe in another PR.

Reply:
Didn't we aggreed to this mocking approach?

Reply:
Yes, nothing wrong with the approach as it. I just annotated that I have some doubt that providing a failMock and a successMock are really needed. I also saw that this is already like that prior to that PR (this is the reason for my [basically unrelated] prefix.

Instead of having several mocks it could be that it makes sense to only have one more generic mock which is able to handle several cases, like we have that e.g. [here](https://github.com/SAP/jenkins-library/blob/master/pkg/mock/runner.go#L22) in a mock we are using for interacting with the shell.